packages/dcos-integration-test/extra/test_legacy_user_management.py

```diff
@@ -0,0 +1,184 @@
+"""
+A collection of tests covering legacy user management in DC/OS.
+
+Legacy user management is considered to be the user management API offered by
+`dcos-oauth` up to DC/OS release 1.12.
+
+Assume that access control is activated in Master Admin Router (could be
+disabled with `oauth_enabled`) and therefore authenticate individual HTTP
+dcos_api_session.
+
+One aspect of legacy DC/OS user management is that once authenticated a user can
+add other users. Unauthenticated HTTP dcos_api_session are rejected by Master
+Admin Router and user management fails (this is the coarse-grained authorization
+model of (open) DC/OS). Here, test that unauthenticated HTTP dcos_api_session
+cannot manage users. However, do not test that newly added users can add other
+users: in this test suite we are limited to having authentication state for just
+a single user available. This is why we can test managing other users only from
+that first user's point of view. That is, we can not test that a user (e.g.
+user2) which was added by the first user (user1) can add another user (user3).
+"""
+import logging
+
+import pytest
+
+from test_helpers import expanded_config
+
+
+__maintainer__ = 'jgehrcke'
+__contact__ = 'security-team@mesosphere.io'
+
+
+log = logging.getLogger(__name__)
+
+
+# Skip entire module in downstream integration tests.
+if 'security' in expanded_config:
+    pytest.skip(
+        'Skip upstream-specific user management tests',
+        allow_module_level=True
+    )
+
+
+def get_users(apisession):
+    r = apisession.get('/acs/api/v1/users')
+    r.raise_for_status()
+    users = {u['uid']: u for u in r.json()['array']}
+    return users
+
+
+def delete_user(apisession, uid):
+    r = apisession.delete('/acs/api/v1/users/%s' % (uid, ))
+    r.raise_for_status()
+    assert r.status_code == 204
+
+
+@pytest.fixture()
+def remove_users_added_by_test(dcos_api_session):
+    users_before = set(get_users(dcos_api_session))
+    log.info('remove_users_added_by_test pre test: users are %s', users_before)
+    try:
+        yield
+    finally:
+        users_after = set(get_users(dcos_api_session))
+        new_uids = users_after - users_before
+        for uid in new_uids:
+            log.info('remove_users_added_by_test post test: remove `%s`', uid)
+            delete_user(dcos_api_session, uid)
+
+
+def test_users_get(dcos_api_session):
```

Comment:
This tests gets run at an arbitrary time. How do we know there are users, and would it be better to test it at a specific time, e.g. maybe as part of the above fixture?